netket/operator/_sumoperators.py

```diff
@@ -0,0 +1,43 @@
+from typing import Optional, List
+
+from netket.utils.types import DType
+
+from netket.hilbert import AbstractHilbert
+from netket.operator import ContinousOperator
+
+import jax.numpy as jnp
+
+
+class SumOperator(ContinousOperator):
+    def __init__(
+        self,
+        hilbert: AbstractHilbert,
+        operators: List,
+        coeff: float,
+        dtype: Optional[DType] = float,
```

Comment:
```suggestion
        *operators: List,
        *
        coefficients: Union[float, List[float]],
        dtype: Optional[DType] = None,
```

Reply:
please write the logic in terms of this itnerface.
coefficients should default to 1. It should be stored as a jnp.array
dtype should use the following formula:

```
        if dtype is None:
            dtype = functools.reduce(
                lambda dt, op: np.promote_types(dt, op.dtype), operators, float
            )
```